### PR TITLE
Lesson03 Exercises Pull Request

### PR DESCRIPTION
**Pull request by CCSt130**
Thank you for your review.

### CODE REVIEW

File: students/CCSt130/lesson03/slicing_lab_ex3-1.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                ("Think in the morning. Act in the noon. Eat in the evening. Sleep in the night...William Blake"),\
+                ("Life without love is like a tree without blossoms or fruit...Khalil Gibran"),\
+                ("No act of kindness, no matter how small, is ever wasted...Aesop"),\
+                ("Love cures people--both the ones who give it and the ones who receive it...Karl A. Menninger"),\
+                ("Dance like no one is watching...Satchel Paige"))
+
+def first_is_last(my_str):
+    """ Manipulate strings in various ways, such as swapping the first and last element. """
+    # Exchange first and last char in str
+    var_swap = my_str[-1:] + my_str[1:-1] + my_str[:+1]
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/slicing_lab_ex3-1.py]: should be just `my_str[:1]` (no `+`)

File: students/CCSt130/lesson03/slicing_lab_ex3-1.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+   
+    return(var_swap)
+
+def every_other(my_str):    
+    # Print str with every other letter removed
+    
+    new_str = my_str[::2]
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/slicing_lab_ex3-1.py]: you can just return new seq directly -> `return seq[::2]`

`my_str` is a poor choice for var name here - your functions should support any sequence: tuple, list, or a string

**Comment by natasha-aleksandrova** [comment]
@CCSt130 based on your comment you are not ready to have your mailroom reviewed - correct?

**Comment by CCSt130** [comment]
Hi Natasha,

Thank you for your email. I apologize, my comments were unclear.

Short story:
My Mailroom Pt. 1 solution is ready to be reviewed. Please do!

Long story:
I left Mailroom Pt. 1 out of my first commit unintentionally, then, after committing it also, I expected it not to be included in the subsequent pull request--Pt. 1 is ready to be reviewed, thanks in advance!

Thank you,
Chuck

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

File: students/CCSt130/lesson03/mail_room_pt1.py [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+        formatted_date = (datetime.date(donations[d][0], donations[d][1], donations[d][2]))
+        # For an array, especially a 2D array, the .format syntax seems easier to implement        
+        # Print date first with 4 spaces on the end
+        print('{}'.format(formatted_date), end = "    ")
+        # Print donor name and amount
+        print('{:<36} ${:>13,.2f}'.format(donations[d][3], donations[d][4]))
+    
+        d+=1 # Increment counter
+
+    print()
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: option # 1 should be printing a report that has following columns and look similar to this:
```
DONOR               | TOTAL GIVEN       | NUMBER OF GIFTS   | AVERAGE GIFTS
-----------------------------------------------------------------------------
James Smith               2527.75                3               842.58
```
it should also be sorted by total given

File: students/CCSt130/lesson03/mail_room_pt1.py [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+            print()
+            print(48 * '*')
+            print()
+            print(17 * '*', "Menu Options", 17 * '*')
+            print()
+
+            # Display menu options
+            print(" 1. Create a Report of all Donations")
+            # print(" 2. Print Thank You Emails for all Donors") TBA
+            print(" 2. Print a List of Donors' Donation Averages")
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: option # 2 should be printing report (shown example output below) sorted by total given

File: students/CCSt130/lesson03/mail_room_pt1.py [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+        while True:
+            # Displaying menu header
+            print()
+            print(48 * '*')
+            print()
+            print(17 * '*', "Menu Options", 17 * '*')
+            print()
+
+            # Display menu options
+            print(" 1. Create a Report of all Donations")
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: option # 1 should be asking for donations to either add or update donors, in addition it should allow the user to use "list" command. (send_thank_you option)

File: students/CCSt130/lesson03/mail_room_pt1.py [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+Created on Sun Jun 16 19:32:15 2019
+"""
+
+import datetime # dates needed for comparison
+
+from collections import Counter
+
+from statistics import mean
+
+if __name__ == "__main__":
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: this is usually at the bottom of the file and `main()` is defined outside of this (this simply executes the main())

File: students/CCSt130/lesson03/mail_room_pt1.py [hunk 7]
```diff
@@ -0,0 +1,500 @@
...
+    print()
+    
+    # Convert enumerated list to tuple and print
+    for names in num_list:
+        donor_list.append(names)
+        print(f' {names[0]}. {names[1]}')
+           
+    print()
+    # Return tuple
+    return(donor_list)
```
Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: none of your functions that are called from my entry point need `return` statements.

Comment (natasha-aleksandrova) [inline students/CCSt130/lesson03/mail_room_pt1.py]: why does this have a return value ?

**Pull request merged**